cirq/ops/fsim_gate.py

```diff
@@ -164,3 +165,268 @@ def __repr__(self) -> str:
 
     def _json_dict_(self) -> Dict[str, Any]:
         return protocols.obj_to_dict_helper(self, ['theta', 'phi'])
+
+
+@value.value_equality(approximate=True)
+class PhasedFSimGate(gate_features.TwoQubitGate,
+                     gate_features.InterchangeableQubitsGate):
+    """General excitation-preserving two-qubit gate.
+
+    The unitary matrix of this gate is:
+
+        [[1,           0,             0,           0],
+         [0,    f1 cos(θ), -i f2 sin(θ),           0],
+         [0, -i f3 sin(θ),    f4 cos(θ),           0],
+         [0,           0,             0, f5 exp(-iφ)]]
+
+    where the phase factors are
```

Comment:
Minor comment: Maybe add some mention that the phase factors aren't independent and can't be individually set in the gate?

Reply:
Done.